docs/core/deploying/preparing-libraries-for-trimming.md

```diff
@@ -0,0 +1,188 @@
+    {
+        // warning IL2067: MyLibrary.Foo(Type): 'type' argument does not satisfy
+        // 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'MyLibrary.UseMethods(Type)'.
+        // The parameter 't' of method 'MyLibrary.Foo(Type)' does not have matching annotations.
+        UseMethods(type);
+    }
+
+    static void UseMethods(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+        Type type)
+    {
+        foreach (var method in type.GetMethods())
+        {
+            // ...
+        }
+    }
+}
+```
+
+Here, `Foo` is calling a method which takes a `Type` argument that is annotated with a [`DynamicallyAccessedMembers`](https://docs.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-5.0) requirement. The requirement states that the value passed in as this argument must represent a type whose public methods are available. In this case, you can fix this by adding the same requirement to the parameter of `Foo`.
+
+```csharp
+    public static void Foo(
+        // Propagate the requirement to Foo's parameter.
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
+        Type type)
+    {
+        // ...
+    }
+```
+
+Like with `RequiresUnreferencedCode`, once you have bubbled up such warnings to public APIs, you are done.
+
+Here is another example where an unknown `Type` flows into the annotated method parameter, this time from a field:
+
+```csharp
+    static Type type;
+
+    static void Bar()
+    {
+        // warning IL2077: MyLibrary.Bar(Type): 'type' argument does not satisfy
+        // 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'MyLibrary.UseMethods(Type)'.
+        // The field 'System.Type MyLibrary::type' does not have matching annotations.
+        UseMethods(type);
+    }
+```
+
+Similarly, here the problem is that the field `type` is passed into a parameter with these requinements. You can fix it by adding `DynamicallyAccessedMembers` to the field. This will warn about code that assigns incompatible values to the field instead. Sometimes this process will continue until a public API is annotated, and other times it will end when a concrete type flows into a location with these requirements. For example:
+
+```csharp
+    [DynamicallyAccessedMembers(DynamicallyAccessedMembers.PublicMethods)]
+    static Type type;
+
+    static void Baz()
+    {
+        MyLibrary.type = typeof(System.Tuple);
+    }
+```
+
+In this case the trim analysis will simply keep public methods of `System.Tuple`, and will not produce further warnings.
+
+### Recommendations
+
+In general, try to avoid reflection if possible, and when using reflection limit it in scope so that it is reachable only from a small part of the library. For example, avoid using non-understood patterns in places like static constructors that will result in the warning propagating to all members of the class.
```

Comment:
```suggestion
In general, try to avoid reflection if possible. When using reflection, limit it in scope so that it is reachable only from a small part of the library. For example, avoid using non-understood patterns in places like static constructors that will result in the warning propagating to all members of the class.
```